src/unwrap.rs

```diff
@@ -0,0 +1,40 @@
+use syntax::ast::*;
+use rustc::lint::{Context, LintPass, LintArray};
+use rustc::middle::ty;
+
+use utils::{span_lint, match_def_path, walk_ty};
+
+#[derive(Copy,Clone)]
+pub struct UnwrapPass;
+
+declare_lint!(pub OPTION_UNWRAP_USED, Warn,
+              "Warn on using unwrap() on an Option value");
+declare_lint!(pub RESULT_UNWRAP_USED, Allow,
+              "Warn on using unwrap() on a Result value");
+
+impl LintPass for UnwrapPass {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(OPTION_UNWRAP_USED, RESULT_UNWRAP_USED)
+    }
+
+    fn check_expr(&mut self, cx: &Context, expr: &Expr) {
+        if let ExprMethodCall(ref ident, _, ref args) = expr.node {
+            if ident.node.name != "unwrap" {
```

Comment:
nit: move the rest of the fn to inside this `if` so that it's visibly easy to extend for other method checks